### ISSUE TITLE
ESLint is still report "'BigInt' is not defined" even ecmaVersion is 2020

### ISSUE THREAD

**Issue by AnnAngela** [eslint/eslint#14320]
**Let me explain first, I need `BigInt` to convert sth, and that cannot be done via literal.**
<hr>

**Tell us about your environment**

* **ESLint Version:**: v7.23.0 (Global, Currently used)
* **Node Version:** v14.16.0
* **npm Version:** v7.6.3
* **Operating System:**  win32 10.0.19042
* **VSCode**: 1.55.1 (user setup)
* **VSCode ESLint extension**: v2.1.19

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

Original one

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "node": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 2020
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
console.info(BigInt(1));
```

*VSCode ESLint extension*

**What did you expect to happen?**
No error.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
Eslint reported "'BigInt' is not defined."
![image](https://user-images.githubusercontent.com/9762652/114484576-1093ec80-9c3d-11eb-893c-db2cf2c5fa40.png)

**Steps to reproduce this issue:**

1. Install VSCode and Eslint extension, as well as eslint installed globally;
2. Clone [AnnAngela/eslint-bigint-reproduce](https://github.com/AnnAngela/eslint-bigint-reproduce)
3. Use VSCode to open the project and open `index.js`.

**Are you willing to submit a pull request to fix this bug?**
No =.=

**Comment by ljharb** [comment]
You still need `env: { es2020: true }` to enable the globals.

**Comment by AnnAngela** [comment]
> You still need `env: { es2020: true }` to enable the globals.

Thx, it resolved my problem, but it made me confused that setting `ecmaVersion` cannot automatically sets `env` but the reverse can.

**Comment by ljharb** [comment]
`ecmaVersion` is about the parser (syntax). `env` is about the environment (API).

I agree it would be nice if ecmaVersion implied env, but it doesn't.

**Comment by AnnAngela** [comment]
> `ecmaVersion` is about the parser (syntax). `env` is about the environment (API).
> 
> I agree it would be nice if ecmaVersion implied env, but it doesn't.

Thx for your kindly explain!
